### DISCUSSION TITLE
aubridge and multiple simultaneous calls

### DISCUSSION THREAD

**Discussion by LukasStolll** [baresip/baresip#2903] [Q&A] [answered]
Hello,

I am using baresip (latest version 3.9.0) under windows as a library with the following use case: 
Establishing a SIP connection with another baresip instance and then "echoing" back the audio from the called peer. 
For that, I just configured aubridge as my source & player, with no other configurations regarding aubridge:

![grafik](https://github.com/baresip/baresip/assets/125663887/bd5b8037-cc41-4a32-997c-e708083bc7fe)

This works absolutely fine, and I get the desired result, listening in on the called peer (another baresip instance) I can hear the audio input (micro or aufile) echoed back. 

However, when making multiple simultaneous calls (in this example two) with the same baresip instance, I only get/hear one audio stream back. 
Looking into the RTP stats from the "echoer", I can see the following:

![grafik](https://github.com/baresip/baresip/assets/125663887/53205a0a-6d11-4aa5-9883-4d4e0a3adb0f)

Trying the same with another audio source & player (for example winwave) booth calls transmit & recive packets and listening in at the peer, I can also hear the echo back multiple times (as expected).

Here is where I have some questions and help would be greatly appreciated:

1. Did I forget any aubridge configuration settings? Is aubridge even capable of briding multiple calls at the same time?
2. Is there another way to echo back the audio? I don't need to play the audio or have an actual source, basically just taking the incoming audio stream and redirect it directly to the outgoing stream. I took a look into the echo module, but aubridge is also used there.

Thanks in advance!

**Comment by sreimers** [comment, marked as answer]
You need to load the `echo` module additionally, it keeps track of handling multiple calls (each connection needs a unique device name for aubridge).

**Reply by LukasStolll** [reply]
Thank you, I will try it out!
